### PR TITLE
ICU-22220 ICU4C APIChangeReport for ICU73

### PR DESCRIPTION
**Pull request by sven-oly**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22220
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/APIChangeReport.md [hunk 1]
```diff
@@ -5,233 +5,460 @@
...
 
 > _Note_ Markdown format of this document is new for ICU 65.
 
-- [Removed from ICU 71](#removed)
-- [Deprecated or Obsoleted in ICU 72](#deprecated)
-- [Changed in  ICU 72](#changed)
-- [Promoted to stable in ICU 72](#promoted)
-- [Added in ICU 72](#added)
-- [Other existing drafts in ICU 72](#other)
+- [Removed from ICU 72](#removed)
```
Comment (richgillam) [inline icu4c/APIChangeReport.md:12]: We can remove stuff that's marked `@stable`?

Comment (richgillam) [reply icu4c/APIChangeReport.md:12]: Actually, on closer examination, it looks like all of these entries show up again below in the "additions" chart.  I think the removal of the `U_OVERRIDE` keyword caused whatever tool is generating this report to not see them as the same function.

**Review by richgillam — CHANGES_REQUESTED** [review]
Sorry to do this, but it looks like there's a lot of old stuff that's showing up as either removed or added (or both), and I'm guessing most or all of those things were neither removed nor added.  A bunch of them seem to be happening because "U_OVERRIDE" was removed from their signatures.  I'm not sure if that's the only problem, though-- I didn't check thoroughly enough to know what's going on.

File: icu4c/APIChangeReport.md (hunk 1 above)
Comment (sven-oly) [reply icu4c/APIChangeReport.md:12]: Thanks for the review. Hmmm, another example of tooling needing updates. Will work on this later today.

Comment (markusicu) [reply icu4c/APIChangeReport.md:12]: I think this is due to this change: https://github.com/unicode-org/icu/commit/a3cbe809091807b406cf526635cb866270a20da2#diff-35fa94c5c8ddb5fa01c3cfb8a0fb1c7f36d6500ad48b0afc0df97660f0c68de4

In StableAPI.java, please change `"[ ]*override", "", // remove override` back to `"[ ]*(override|U_OVERRIDE)", "", // remove U_OVERRIDE and override`, and then regenerate. (Make sure to copy-paste this exactly.)

**Comment by sven-oly** [comment]
After updating the .java file (included), the API change reports look correct. Thanks, Markus, for finding this.

Ready for another review!

File: icu4c/APIChangeReport.md (hunk 1 above)
Comment (sven-oly) [reply icu4c/APIChangeReport.md:12]: Update done to StableAPI.java, fixing the reports.

File: tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java [hunk 2]
```diff
@@ -552,8 +552,7 @@ private static String getBasename(String str) {
                                       //  TODO: notify about this difference, separately
                 "[ ]*U_NOEXCEPT", "", // remove U_NOEXCEPT (this was fixed in Doxyfile, but fixing here so it is
                                       //  retroactive)
-                "[ ]*override", "",   // remove override
-                // Simplify possibly-covariant functions to void*
+                "[ ]*(override|U_OVERRIDE)", "", // remove U_OVERRIDE and override                // Simplify possibly-covariant functions to void*
                 "^([^\\* ]+)\\*(.*)::(clone|safeClone|cloneAsThawed|freeze|createBufferClone)\\((.*)", "void*$2::$3($4",
```
Comment (markusicu) [inline tools/release/java/src/main/java/com/ibm/icu/dev/tools/docs/StableAPI.java:556]:
```suggestion
                "[ ]*(override|U_OVERRIDE)", "", // remove U_OVERRIDE and override
                // Simplify possibly-covariant functions to void*
                "^([^\\* ]+)\\*(.*)::(clone|safeClone|cloneAsThawed|freeze|createBufferClone)\\((.*)", "void*$2::$3($4",
```

File: icu4c/APIChangeReport.html [hunk 3]
```diff
@@ -38,672 +38,547 @@ <h1>ICU4C API Comparison: ICU 71 with ICU 72</h1>
...
+<th>File</th><th>API</th><th>ICU 72</th><th>ICU 73</th>
 </tr>
 </THEAD>
 </table>
 <P></P>
 <a href="#_top">(jump back to top)</a>
 <hr>
 <a name="changed"></a>
-<h2>Changed in  ICU 72 (old, new)</h2>
+<h2>Changed in  ICU 73 (old, new)</h2>
```
Comment (richgillam) [inline icu4c/APIChangeReport.html:77]: All the same stuff shows up both under "Changed in ICU 73" and "Promoted to stable in ICU 73".  I wonder if there's a way to filter the "promoted to stable" stuff out of the "changed" list?

**Review by richgillam — APPROVED** [review]
Oh, this looks SO much better!  Thank you for the change!  There is still one thing that looks a little weird, but I'm fine if you ignore it for now and leave it for some future release.

File: icu4c/APIChangeReport.html (hunk 3 above)
Comment (markusicu) [reply icu4c/APIChangeReport.html:77]: I am sure that we have a ticket or two for simplifying the API change report. Probably filed by me...

**Review by markusicu — COMMENTED** [review]
lgtm pse squash

**Comment by sven-oly** [comment]
Please take another look - I'll submit once I get an LGTM.

**Review by richgillam — APPROVED** [review]
Still looks good.

**Pull request merged**